src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -356,7 +395,11 @@ public Survey getSurveyMostRecentlyPublishedVersion(StudyIdentifier studyIdentif
      * version (not a specific timestamped version), this method should be rarely called.
      */
     @Override
-    public Survey getSurvey(GuidCreatedOnVersionHolder keys, boolean includeElements) {
+    public Survey getSurvey(StudyIdentifier studyIdentifier, GuidCreatedOnVersionHolder keys, boolean includeElements) {
+        if (keys.getGuid() != null && keys.getGuid().toLowerCase().startsWith(IDENTIFIER_PREFIX)) {
+            return new QueryBuilder().setStudy(studyIdentifier).setSurvey(keys.getGuid()).setCreatedOn(keys.getCreatedOn())
+                    .setSkipElements(!includeElements).getOne();
+        }
```

Comment:
This duplicates a lot of the same code as getAll(). Why not always pass in studyIdentifier?

I would go as far as to put the null-check in QueryBuilder.setStudy(). If the object studyIdentifier is null, then it sets the string studyIdentifier to null.